### PR TITLE
Interactive Star Rating Component

### PR DESCRIPTION
**Pull request by gunjankothari**

### CODE REVIEW

File: packages/core/src/components/StarRating.tsx [hunk 1]
```diff
@@ -62,7 +91,7 @@ export const SEED_DATA = {
   name: "Star Rating",
   tag: "StarRating",
   description: "A star rating component",
-  category: COMPONENT_TYPES.basic,
+  category: COMPONENT_TYPES.button,
   props: {
     starSize: createStaticNumberProp({
```
Comment (dhruska) [inline packages/core/src/components/StarRating.tsx:96]: To make this work with Draftbit's state system, you'll also want to remove the "rating" prop below and add (I think):

```suggestion
    fieldName: createFieldNameProp({
      defaultValue: "ratingValue", // this is the name of the variable declared on the screen in Draftbit
      handlerPropName: "onPress", // the change handler prop in this component
      valuePropName: "rating", // the value prop in this component
    }),
    starSize: createStaticNumberProp({
```

File: packages/core/src/components/StarRating.tsx [hunk 2]
```diff
@@ -54,6 +73,16 @@ const styles = StyleSheet.create({
...
   },
+  touchContainer: {
+    display: "flex",
+    flexDirection: "row",
+    position: "absolute",
+    top: 0,
+    right: 0,
+    left: 0,
+    bottom: 0,
+    zIndex: 111,
```
Comment (dhruska) [inline packages/core/src/components/StarRating.tsx:84]: Is it possible to make this a more reasonable value like `1` or `2`?

Comment (gunjankothari) [reply packages/core/src/components/StarRating.tsx]: Fixed

File: packages/core/src/components/StarRating.tsx (hunk 1 above)
Comment (gunjankothari) [reply packages/core/src/components/StarRating.tsx:96]: updated

**Review by dhruska — APPROVED** [review]

**Pull request merged**